### PR TITLE
Add util function to disable all remote methods of a model

### PR DESCRIPTION
**Pull request by haishanh**
Not sure if `server/lib/utils.js` is the best place to add this.

### CODE REVIEW

**Comment by makara** [comment]
- I prefer making `utils` a dir so it can be easier extended or skipped, i.e `lib/utils/disableAllRemoteMethods.js`.
- I'd name it `disableRemoteMethods` and give it an argument for a whitelist (to be not disabled).
- Some doc/example/test would be great.

**Comment by haishanh** [comment]
Did `git mv server/lib/utils.js server/utils/disableRemoteMethods.js`, and renamed `disableAllRemoteMethods` to `disableRemoteMethods`, argument `excepts`(the *whitelist*) to `allowedMethods`.

I think the JSDoc is pretty clean.

If I add a test, I'll need to define a model, which will need a data source i.e. introducing a lot of extra stuff into the repo.

Comments and ideas are welcome.

**Comment by paulvollmer** [comment]
i think it make sense to define a simple model because most time we use loopback with some models. if we have a more 'real world' boilerplate it's faster to bootstrap other projects.

as datasource we can use the simple 'memory' connector...

/cc @haishanh @makara

File: server/utils/disableRemoteMethods.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * @param {array<string>} allowedMethods - list of allowed remote method names
+ * @return {undefined}
+ */
+module.exports = function disableRemoteMethods(model, allowedMethods = []) {
+  model.sharedClass.methods().forEach(method => {
+    if (allowedMethods.indexOf(method.name) < 0) {
+      if (method.isStatic) {
+        model.disableRemoteMethodByName(method.name);
+      } else {
+        model.disableRemoteMethodByName('prototype.' + method.name);
```
Comment (xavierchow) [inline server/utils/disableRemoteMethods.js]: Why  not `MyModel.disableRemoteMethod('deleteById', isStatic);`?
https://loopback.io/doc/en/lb2/Exposing-models-over-REST.html#hiding-methods-and-rest-endpoints

Or am I looking into a old spec?

Comment (haishanh) [reply server/utils/disableRemoteMethods.js]: it's indeed old spec, lastest is https://loopback.io/doc/en/lb3/Exposing-models-over-REST.html#hiding-methods-and-rest-endpoints

Comment (xavierchow) [reply server/utils/disableRemoteMethods.js]: 👍

**Comment by ole3021** [comment]
also implemented in PR #40

**Pull request merged**